src/rail_gui.cpp

```diff
@@ -1308,18 +1299,37 @@ struct BuildRailStationWindow : public PickerWindowBase {
 		}
 	}
 
+	/**
+	 * Get dimensions in pixels of station type names for the given width.
+	 * @param width Width of text in pixels.
+	 * @returns Dimensions required for station type name.
+	 */
+	Dimension GetStationTypeNameDimension(int width) const
+	{
+		int y = 0;
+		for (StationClassID cls_id = STAT_CLASS_BEGIN; cls_id < StationClass::GetClassCount(); cls_id++) {
+			if (cls_id == STAT_CLASS_WAYP) continue;
+			const StationClass *stclass = StationClass::Get(cls_id);
+			for (uint j = 0; j < stclass->GetSpecCount(); j++) {
+				const StationSpec *statspec = stclass->GetSpec(j);
+				StringID str = (statspec != nullptr && statspec->name != 0) ? statspec->name : STR_STATION_CLASS_DFLT_STATION;
```

Comment:
Maybe move this logic into its own function, because it's used multiple times and it's not that trivial.